sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_repository_client.py

```diff
@@ -123,25 +126,95 @@ def list_registry_artifacts(self, **kwargs):
-            last=last,
-            n=n,
-            orderby=orderby,
-            cls=lambda objs: [
-                RegistryArtifactProperties._from_generated(x) for x in objs  # pylint: disable=protected-access
-            ],
+        cls = kwargs.pop('cls', lambda objs: [RegistryArtifactProperties._from_generated(x) for x in objs])
+
+        error_map = {
+            401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError
+        }
+        error_map.update(kwargs.pop('error_map', {}))
+        accept = "application/json"
+
+        def prepare_request(next_link=None):
+            # Construct headers
+            header_parameters = {}  # type: Dict[str, Any]
+            header_parameters['Accept'] = self._client._serialize.header("accept", accept, 'str')
+
+            if not next_link:
+                # Construct URL
+                url = '/acr/v1/{name}/_manifests'
+                path_format_arguments = {
+                    'url': self._client._serialize.url("self._client._config.url", self._client._config.url, 'str', skip_quote=True),
+                    'name': self._client._serialize.url("name", name, 'str'),
+                }
+                url = self._client._client.format_url(url, **path_format_arguments)
+                # Construct parameters
+                query_parameters = {}  # type: Dict[str, Any]
+                if last is not None:
+                    query_parameters['last'] = self._client._serialize.query("last", last, 'str')
+                if n is not None:
+                    query_parameters['n'] = self._client._serialize.query("n", n, 'int')
+                if orderby is not None:
+                    query_parameters['orderby'] = self._client._serialize.query("orderby", orderby, 'str')
+
+                request = self._client._client.get(url, query_parameters, header_parameters)
+            else:
+                url = next_link
+                query_parameters = {}  # type: Dict[str, Any]
+                path_format_arguments = {
+                    'url': self._client._serialize.url("self._client._config.url", self._client._config.url, 'str', skip_quote=True),
+                    'name': self._client._serialize.url("name", name, 'str'),
+                }
+                url = self._client._client.format_url(url, **path_format_arguments)
+                request = self._client._client.get(url, query_parameters, header_parameters)
+            return request
+
+        def extract_data(pipeline_response):
+            deserialized = self._client._deserialize('AcrManifests', pipeline_response)
+            list_of_elem = deserialized.manifests
+            if cls:
+                list_of_elem = cls(list_of_elem)
+            link = None
+            if "Link" in pipeline_response.http_response.headers.keys():
+                link = _parse_next_link(pipeline_response.http_response.headers["Link"])
+            return link, iter(list_of_elem)
```

Comment:
just to clarify: i remember you said before that the first part of the raw link in the response headers includes what you need to pass as a query param to next calls. It seems you're ignoring that info rn, want to check to see if that's right

Reply:
The next URL to use is parsed from the `Link` header in the response. If it is present in the headers we parse it here. Does that answer your question?

Reply:
I was just recalling when we talked about this in our video chat, that you said the `Link` header contains _both_ the next link you want to make your call against _and_ additional information you wanted to pass to the next call (i believe you said some query parameters, but not sure). Asked this question because it seems like the only information you're getting out of the `Link` header is the next link, _not_ the additional information you want to pass to the next call

Reply:
I must have misspoke, the Link is pre-formatted with the query parameters so all we need to do is get the link, there's no other info necessary in the headers.